src/storage/v2/durability/durability.cpp

```diff
@@ -30,6 +30,7 @@
 #include "storage/v2/durability/wal.hpp"
 #include "storage/v2/inmemory/label_index.hpp"
 #include "storage/v2/inmemory/label_property_index.hpp"
+#include "storage/v2/inmemory/storage.hpp"
```

Comment:
Either:
durability includes inmemory storage OR
inmemory storage includes durability
not both

Reply:
But I think I cannot forward declare `InMemoryStorage` since I need to access its methods and variables?

Reply:
How should I do it?

Reply:
You was already passing the necessary data members (de-structured before the call). You changed it to use `InMemoryStorage *`...you didn't need to do that change.

Reply:
I opted for reverting changes here. The main reason is that I could couple only few arguments but not all hence whether I send 9 or 6 arguments is almost the same.